components/GitHubEventAction.tsx

```diff
@@ -1,10 +1,26 @@
+import { CreateEvent, DeleteEvent, ForkEvent, IssueCommentEvent, IssuesEvent, MemberEvent,
+  PullRequestEvent, PullRequestReviewCommentEvent, PullRequestReviewEvent,
+  PushEvent, PublicEvent, WatchEvent } from '@octokit/webhooks-types';
 import React from 'react';
 import ELink from './ELink';
-// TODO(mattxwang): fix the payload thing to actually use a type, maybe
-// from the octokit types
+
+type payloadType =
+| CreateEvent
+| DeleteEvent
+| ForkEvent
+| IssueCommentEvent
+| IssuesEvent
+| MemberEvent
+| PullRequestEvent
+| PullRequestReviewCommentEvent
+| PullRequestReviewEvent
+| PushEvent
+| PublicEvent
+| WatchEvent;
```

Comment:
This is great!

One thing I'll tag on - in the switch-case below, you may be able to take advantage of automatic [Type Narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html); the section on [Discriminated unions](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) is probably the most relevant.. Instead of having to narrow the payload by casting the type (which is "technically" potentially dangerous), you can switch-case off of the actual value within the union.

(the caveat is that TS's type inference seems slightly worse than Flow's, so I'm not sure if the compiler will be "smart" enough to figure this out. if not, then don't worry about this!)